test/BreakEventHandler.js

```diff
@@ -0,0 +1,93 @@
+var expect = require('chai').expect,
+    launcher = require('./helpers/launcher'),
+    InjectorClient = require('../lib/InjectorClient').InjectorClient,
+    BreakEventHandler = require('../lib/BreakEventHandler').BreakEventHandler,
+    ScriptManager = require('../lib/ScriptManager').ScriptManager;
+    
+describe('BreakEventHandler', function() {
+  var childProcess,
+      session;
+
+  describe('should cache events', function() {
+    it('if started in debug-brk mode', function(done) {
+      prepareBreakEventHandler(true, function() {
+        expect(session.breakEventHandler._eventsCache).to.instanceof(Array);
+        expect(session.breakEventHandler._eventsCache.length).to.equal(0);
+        
+        session.debuggerClient.request('continue', undefined, function() {
+          childProcess.stdin.write('pause\n');
+          setTimeout(function() {
+            expect(session.breakEventHandler._eventsCache.length).to.equal(1);
+            done();
+          }, 100);
+        });
+      });
+    });
+    
+    it('if started in debug mode', function(done) {
+      prepareBreakEventHandler(false, function() {
+        expect(session.breakEventHandler._eventsCache).to.instanceof(Array);
+        expect(session.breakEventHandler._eventsCache.length).to.equal(0);
+        
+        childProcess.stdin.write('pause\n');
+        setTimeout(function() {
+          expect(session.breakEventHandler._eventsCache.length).to.equal(1);
+          done();
+        }, 100);
+      });
+    });
+  });
+  
+  describe('should resolve events', function() {
+    before(function(done) {
+      prepareBreakEventHandler(false, function() {
+        childProcess.stdin.write('pause\n');
+        setTimeout(done, 100);
+      });
+    });
+    
+    it('on resource-tree-resolved', function(done) {
+      session.frontendClient.once('Debugger.paused', function() {
+        done();
+      });
+      
+      session.debuggerClient.request('scripts', null, function(error, result) {
+        if (error) done(error);
+
+        result.forEach(session.scriptManager.addScript.bind(session.scriptManager));
+        session.resourceTreeResolved = true;
+        session.emit('resource-tree-resolved');
+      });
+    });
```

Comment:
This is a main test for #534, so I test what we have consistent state and `BreakEventHandler` 
doesn't ignore events [here](https://github.com/3y3/node-inspector/commit/27dd9cabd239c60c748d77d9d1f27adacec47787#diff-3cf7bfb8d4bd4a7cd3212e0854ee8523R96)

```
describe('event resolver', function() {
  it('should send cached events to frontend on `resource-tree-resolved`');
  it('should send events to frontend after `resource-tree-resolved`');
});
```

It meybe reasonable to check `eventsCache.length` here.